todo_list_app/views.py

```diff
@@ -0,0 +1,221 @@
+            login(request, user)
+            return redirect('index')
+        except LoginError as e:
+            messages.error(request, str(e))
+            return redirect('login')
+    else:
+        form = LoginForm()
+        return render(request, 'auth/login.html', {'form': form})
+
+
+def logout_user(request):
+    logout(request)
+    messages.success(request, 'You have been logged out.')
+    return redirect('index')
+
+
+def register_user(request):
+    if request.method == 'POST':
+        form = CustomUserCreationForm(request.POST)
+        if form.is_valid():
+            form.save()
+            username = form.cleaned_data.get('username')
+            password = form.cleaned_data.get('password1')
+            user = authenticate(username=username, password=password)
+            login(request, user)
+            messages.success(request, 'You have been registered successfully.')
+            return redirect('index')
+    else:
+        form = CustomUserCreationForm()
+
+    return render(request, 'auth/register.html', {'form': form})
+
+
+def index(request):
+    return render(request, 'todo_list_app/index.html')
+
+
+@login_required
+def all_categories(request):
+
+    query = request.GET.get('q')
+    category_list = Category.objects.filter(user=request.user).values('id', 'name')
+    if query:
+        category_list = category_list.filter(name__icontains=query)
+    return render(request, 'todo_list_app/all_categories.html', {'category_list': category_list,'query':query})
+
+
+@login_required
+def delete_category(request, category_id):
+    Category.objects.get(id=category_id, user=request.user).delete()
+    return redirect('todo_list:all_categories')
+
+
+@login_required
+def update_category(request, category_id):
+    category = Category.objects.get(id=category_id, user=request.user)
+    if request.method == 'POST':
+        name = request.POST.get('name')
+        category.name = name
+        category.save()
+        return redirect('todo_list:all_categories')
+    else:
+        return render(request, 'todo_list_app/category_update.html', {'category': category})
+
+
+@login_required
+def create_category(request):
+    if request.method == 'POST':
+        form = CategoryForm(request.POST)
+        if form.is_valid():
+            category = form.save(commit=False)
+            category.user_id = request.user.id
+            category.save()
+            return redirect('todo_list:all_categories')
+    else:
+        form = CategoryForm()
+        return render(request, 'todo_list_app/category_create.html', {'form': form})
+
+
+def local_to_utc(request,reminder):
+    user_timezone = request.POST.get('timezone')
+    if user_timezone:
+        user_tz = pytz.timezone(user_timezone)
+    else:
+        user_tz = pytz.utc
+    reminder.remind_at = reminder.remind_at.replace(tzinfo=None)
+    reminder.remind_at = user_tz.localize(reminder.remind_at, is_dst=None)
```

Comment:
This function is a utility. Create a file named `utility.py` and move this function to it.